### PR TITLE
Add city and country validation at app side.

### LINKED ISSUE #1131: Creating patient without adding mandatory field city and country
**Describe the bug**
Able to add patient without adding mandatory field city and country

**Component**
Demo App

**To Reproduce**
1. Go to 'Demo App'
2. Click on '+'
3. Add all patient (mandatory fields) details except city and country and click on submit
4. Observe behavior

**Expected behavior**
Patient should not get created without adding city and country(mandatory field)

**video**
https://user-images.githubusercontent.com/64417713/152964847-8cbc7f2b-5ae9-4e11-824b-e0857a6f0aa1.mp4

**Smartphone (please complete the following information):**
 - Device: Lenovo tab
 - Android version: 10

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1131 

**Description**
City and Country validation is added.
If values are blank then it will not create patient.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by santosh-pingle** [comment]
> i'm not sure this is the right place to fix this issue. i would much rather that the sdc lib return some kind of result.
> 
> can you try to use this api:
> 
> https://github.com/google/android-fhir/blob/efcdabb7ae179ce4c90246c465b271a78479dd87/datacapture/src/main/java/com/google/android/fhir/datacapture/validation/QuestionnaireResponseValidator.kt#L100

I see mentioned api throws exception if one of the check fails. I am thinking to add `required` check, if fails then throw exception.

**Comment by jingtang10** [comment]
just to reiterate what we discussed before, the check should be in the `validateQuestionnaireResponse` api rather than `checkQuestionnaireResponse`.

File: demo/src/main/java/com/google/android/fhir/demo/AddPatientViewModel.kt [hunk 1]
```diff
@@ -53,23 +54,26 @@ class AddPatientViewModel(application: Application, private val state: SavedStat
...
+        )
+        .values
+        .flatten()
+        .filterNot { it.isValid }
+        .let {
+          if (it.isNotEmpty()) {
+            isPatientSaved.value = false
+            return@launch
+          }
+        }
```
Comment (jingtang10) [inline demo/src/main/java/com/google/android/fhir/demo/AddPatientViewModel.kt:70]:
```suggestion
        .any { !it.isValid }
```
and wrap this in an `if` statement to return

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Review by jingtang10 — APPROVED** [review]
Thanks @santosh-pingle

**Pull request merged**